### DISCUSSION TITLE
Beginner Question: How to send Base64 encoded JPG from Make to Dify and process it with LLM?

### DISCUSSION THREAD

**Discussion by ozz-hand** [langgenius/dify#8091] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Hello,

I’m relatively new to Make and Dify, and I’m trying to integrate the two for a project. Specifically, I want to send a JPG image from Make to Dify, but I’m not sure about the correct way to do this. In Make, I plan to convert the JPG to Base64 before sending it.

I have a couple of beginner-level questions:

1. Is it possible to send a Base64-encoded JPG from Make to Dify using an HTTP module? If so, are there any specific settings or parameters I should be aware of?

2. Will Dify be able to handle the Base64-encoded JPG and process it with LLM? I want to make sure Dify can decode the image and run the necessary processes on it.

I’d appreciate any advice or guidance, as I’m still learning!

Thank you in advance for your help!

**Comment by ozz-hand** [comment]
**Update**
After further investigation, I realized the issue was on my end. I had overlooked a setting in Dify—there’s an option to enable image input, and in my case, it was disabled. This was the reason why, even though I was sending the image data via the API, it wasn’t being recognized or stored in the sys.files field.

**Comment by mcgrimm** [comment]
Hi, ozz-hand , i also met such situation , i want to know if or how can i send base64 to dify to be able process in LLM , can you please let me know what option and which LLM you were using? thank you very much.

**Reply by ozz-hand** [reply]
Hello, McGrimm. I hadn't noticed this option before.
![image](https://github.com/user-attachments/assets/f7efe656-888c-444a-ab85-50cab3b0c743)

It seems this option is going to become obsolete because sys.file will be replaced by another function in the future.
I am currently using GPT-4o. This is because I need an LLM that supports image inputs.

**Comment by wow-ww** [comment]
Hi!I get image base64 by Tool.How to send base64 to LLM node?Could u help me?

**Reply by mcgrimm** [reply]
I write javascript code to simulate the image upload function. first convert the base64 to blob , then upload the blob to dify by file upload api , finally your got the file local id , using local id you can trigger the llm api.

**Reply by wow-ww** [reply]
Thanks!Do you mean to write an API locally to convert the base64-encoded image into a blob and call it through an http node? I have encountered a new problem: I wrote an API locally, and I will call it through an http node in dify. I hope that this API will return multiple image files, and then give the returned multiple image files to the LLM Node. I don't know how to return multiple files.
